components/brave_rewards/browser/rewards_service.cc

```diff
@@ -30,6 +56,25 @@ void RewardsService::RemoveObserver(RewardsServiceObserver* observer) {
   observers_.RemoveObserver(observer);
 }
 
+std::string RewardsService::GetPublisherIdFromURL(const GURL& url) {
+  if (IsGreaselionURL(url)) {
+    return "";
```

Comment:
use base::optional instead of returning an empty string. It was also a bit confusing to me why we had a greaselion check here and I think we should use some other term for these. Don't we already have some term for sites that support multiple creators within them?

Reply:
similar to the context thing, I would also separate this from the RewardsService itself (some kind of util file?) for dependency reasons